src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -1083,131 +997,140 @@ public boolean remove(Runnable task) {
     }
 
     /**
-     * Adds a task for immediate execution.  No safety checks are done at this point, the task 
-     * will be immediately added and available for consumption.
+     * Call to get the total quantity of tasks within both stored queues.  This returns the total 
+     * quantity of items in both the execute and scheduled queue.  If there are scheduled tasks 
+     * which are NOT ready to run, they will still be included in this total.
      * 
-     * @param task Task to add to end of execute queue
+     * @return Total quantity of tasks queued
      */
-    public void addExecute(OneTimeTaskWrapper task) {
-      executeQueue.add(task);
-
-      handleQueueUpdate();
+    public int queueSize() {
+      return executeQueue.size() + scheduleQueue.size();
     }
 
-    /**
-     * Adds a task for delayed execution.  No safety checks are done at this point.  This call 
-     * will safely find the insertion point in the scheduled queue and insert it into that 
-     * queue.
-     * 
-     * @param task Task to insert into the schedule queue
-     */
-    public void addScheduled(TaskWrapper task) {
+    public void drainQueueInto(List<TaskWrapper> removedTasks) {
+      clearQueue(executeQueue, removedTasks);
       synchronized (scheduleQueue.getModificationLock()) {
-        ClockWrapper.stopForcingUpdate();
-        try {
-          int index = ListUtils.getInsertionEndIndex(scheduleQueue, task, true);
-          
-          scheduleQueue.add(index, task);
-        } finally {
-          ClockWrapper.resumeForcingUpdate();
-        }
+        clearQueue(scheduleQueue, removedTasks);
       }
-      
-      handleQueueUpdate();
     }
-
-    /**
-     * Adds a scheduled task to the end of the scheduled queue.  It is expected that this task is 
-     * NOT ready for execution, and will later be moved from invoking 
-     * {@link #reschedule(RecurringTaskWrapper)}.
-     * 
-     * @param task Task to add to end of schedule queue
-     */
-    public void addScheduledLast(RecurringTaskWrapper task) {
-      scheduleQueue.addLast(task);
-      // no need to notify since this task wont be ready to run
+  
+    private static void clearQueue(Collection<? extends TaskWrapper> queue, List<TaskWrapper> resultList) {
+      Iterator<? extends TaskWrapper> it = queue.iterator();
+      while (it.hasNext()) {
+        TaskWrapper tw = it.next();
+        tw.cancel();
+        if (! (tw.task instanceof ShutdownRunnable)) {
+          int index = ListUtils.getInsertionEndIndex(resultList, tw, true);
```

Comment:
I now added the feature that on shutdown the returned task list is ordered roughly in the order of execution....before it was basically a completely random list.